src/Command/InfectionCommand.php

```diff
@@ -362,7 +362,7 @@ private function runConfigurationCommand(Locator $locator): void
             $locator->locateOneOf(InfectionConfig::POSSIBLE_CONFIG_FILE_NAMES);
         } catch (\Exception $e) {
             $configureCommand = $this->getApplication()->find('configure');
-            $config = $this->getContainer()->get('infection.config');
+            $config = $this->getContainer()->get('infection.config.initial');
```

Comment:
Seems like the only one reason why `$config` variable is used here is to get default test framework name (`$config->getTestFramework()`) when `infection.json` **does not exist**.

And since config does not exist, it always defaults to `phpunit`.

https://github.com/infection/infection/blob/5b02e8e3c77b10732e69e7aad005946d06a431ad/src/Config/InfectionConfig.php#L155

`$this->config->testFramework` is empty when `infection.json` does not exist. Because of

https://github.com/infection/infection/blob/5b02e8e3c77b10732e69e7aad005946d06a431ad/src/Config/ConfigCreatorFacade.php#L80-L85

If we replace `$config->getTestFramework()` with `TestFrameworkTypes::PHPUNIT`, this will be the same PR as https://github.com/infection/infection/pull/577

I think we don't need `$config` instance here at all.

Reply:
I've seen this done in #577, but decided to keep the current behavior. For one because if configuration values are coming from one source, then you don't have to walk around all code changing them. Change them in one place and be done. For another we may have an idea to guess the value in the future. Then, again, everything will be as ready for this as it is. 

I think some of these reasons are behind why this was done like that in the first place.

Do you still think we should stick to PHPUnit defaulting with a plain constant?

Reply:
Apologize for the late reply.

What you say is perfectly valid. The only issue with this particular line of code is that it was my mistake (read "bug") that I've added it here. Let me explain:

This is a commit that adds this config here: https://github.com/infection/infection/commit/2859730b9ffa5678e3a90226cae0d1647f0ff8f6

From git commit message:

> * Use default test framework if neither CLI nor config has the value for it

So, all I wanted to do, is to use **default** test framework when it's impossible to get it neither  from CLI option nor from the config. 

Why it was a mistake? Because `$config->getTestFramework()` is intended return value from `infection.json`, not a default TF. It just has a fallback.

What I think would be the right solution:

instead of calling instance level method `->getTestFramework()`, I would introduce a new static method `::getDefaultTestFramework()` and call it here statically.

1. It would resolve the issue with config initialization, because of a statically called method
2. It would make code more expressive because what we want to do here is really get default TF

Reply:
OK, I got what you're thinking. This makes sense.

Reply:
I went with `TestFrameworkTypes::PHPUNIT` for now.

(We can make an easy switch to `InfectionConfig::getDefaultTestFramework()` when it'll do anything in particular other than just returning `TestFrameworkTypes::PHPUNIT`.)